Refactor in core.

cobward opened the pull request.

justAnIdentity committed:
OID4VP ID2 - WIP

wip - add id2 mdoc request support

id2 WIP

fix mdl path matching

oid4vp id2 WIP

OID4VP ID2

use ssh for isomdl dep

clean up

update isomdl version

justAnIdentity committed:
various fixes and updates

justAnIdentity committed:
clippy fix

justAnIdentity committed:
small fixes

justAnIdentity committed:
clippy fix

justAnIdentity committed:
remove openssl

justAnIdentity committed:
clean up

justAnIdentity committed:
clippy fix

Commit:
Use josekit fork

Commit:
Update josekit dep

Commit:
Fix dependencies

justAnIdentity committed:
Send impl for async trait

justAnIdentity committed:
Merge branch 'feat/oidc4vp-ID2' of github.com:spruceid/oidc4vp-rs into feat/oidc4vp-ID2

justAnIdentity committed:
fix unwraps

justAnIdentity committed:
propagate error message for empty error

justAnIdentity committed:
fix presentation_submission

justAnIdentity committed:
fmt

Commit:
Refactor in core.

Commit:
Update isomdl dep

Commit:
Clippy

Commit:
Make client_metadata inner public

sbihel commented on Cargo.toml:
```diff
@@ -9,29 +9,28 @@ repository = "https://github.com/spruceid/oidc4vp-rs/"
+async-trait = "0.1.73"
+base64 = "0.21.4"
+did-web = "0.2.2"
+didkit = "0.6.0"
```
I don't think libraries should use didkit, the APIs should instead accept a `DIDResolver` and let the downstream consumer choose what DIDs to support

sbihel commented:
Could you take this opportunity to rename the crate to `oid4vp` as well?

Commit:
Remove didkit, rename crate, add missing metadata.

Commit:
Various updates

- rename crate
- make http client to be configurable
- refactor traits and request builder
- add client metadata resolution

Commit:
Add VpFormats metadata param

sbihel commented on src/core/authorization_request/parameters.rs:
```diff
@@ -0,0 +1,508 @@
+                ));
+        }
+
+        bail!("")
```
https://github.com/spruceid/oid4vp-rs/pull/4

justAnIdentity commented:
@cobward this looks great. Finally had a look at it 🙈

I'm planning an update and refactor of isomdl-18013-7.
There will probably be some todo's on here that spill out from that update.
Do you want to merge this and have new issues created to take care of those, or would you rather include them in this refactor in one go?
